### PR TITLE
OOP School Library: Creating Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by aima98**
### For this milestone, I implemented the following features :

- [ ]  Creating classes Person, Student And Teacher.
- [ ]  Adding Linters.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(classroom, age, name = 'unknown', parent_permission: true)
+    super(age, name, parent_permission)
```
Comment (TedLivist) [inline student.rb]: - [ ] The `parent_permission` uses colon notation, so its application as an argument in the `#super` method should be slightly different.

  ```
  super(age, name, parent_permission: parent_permission)
  ```

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(specialization, age, name = 'unknown', parent_permission: true)
+    super(age, name, parent_permission)
```
Comment (TedLivist) [inline teacher.rb]: - [ ] The `parent_permission` uses colon notation, so its application as an argument in the `#super` method should be slightly different.

  ```
  super(age, name, parent_permission: parent_permission)
  ```

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  def of_age?
+    @age >= 18
+  end
+
+  public
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
```
Comment (TedLivist) [inline person.rb:22]: - [ ] **[OPTIONAL]** Instead defining `private` and `public`, it is often the convention that private methods are put at the bottom of the class, so there is no need to define `private` and then `public` methods again.

  ```
  def can_use_services?
  ...

  private

  def of_age?
  ...
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @aima98,

Good job so far!
✔️ Classes are different files based on convention
✔️ Implementation of class inheritance
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @aima98,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**